tests/pytests/unit/modules/test_network.py

```diff
@@ -0,0 +1,114 @@
+
+
+@pytest.fixture
+def expected_routes():
+    return [
+        {
+            "destination": "0.0.0.0",
+            "addr_family": "inet",
+            "netmask": "0.0.0.0",
+            "flags": "UG",
+            "interface": "eth0",
+            "gateway": "100.64.24.145",
+        },
+        {
+            "destination": "100.64.24.144",
+            "addr_family": "inet",
+            "netmask": "255.255.255.240",
+            "flags": "U",
+            "interface": "eth0",
+            "gateway": "0.0.0.0",
+        },
+        {
+            "destination": "169.254.0.0",
+            "addr_family": "inet",
+            "netmask": "255.255.0.0",
+            "flags": "U",
+            "interface": "eth0",
+            "gateway": "0.0.0.0",
+        },
+        {
+            "destination": "192.168.0.0",
+            "addr_family": "inet",
+            "netmask": "255.255.255.0",
+            "flags": "UG",
+            "interface": "eth0",
+            "gateway": "100.64.24.145",
+        },
+        {
+            "destination": "192.168.1.0",
+            "addr_family": "inet",
+            "netmask": "255.255.255.0",
+            "flags": "UG",
+            "interface": "eth0",
+            "gateway": "100.64.24.145",
+        },
+    ]
+
+
+def test_ip_route_linux_ipv4(expected_routes):
+    """
+    Tests that _ip_route_linux can parse ipv4 routes appropriately
+    """
+    ip_route_output = [
+        "default via 100.64.24.145 dev eth0",
+        "100.64.24.144/28 dev eth0 proto kernel scope link src 100.64.24.148",
+        "169.254.0.0/16 dev eth0 scope link metric 1002",
+        "192.168.0.0/24 via 100.64.24.145 dev eth0",
+        "192.168.1.0/24 via 100.64.24.145 dev eth0",
+    ]
+
+    mock_cmd = MagicMock(return_value=os.linesep.join(ip_route_output))
+    mock_path = MagicMock(return_value=False)
+    with patch.dict(network.__grains__, {"kernel": "Linux"}):
+        with patch.dict(
+            network.__utils__,
+            {"path.which": mock_path, "network.calc_net": utils_network.calc_net},
+        ):
+            with patch.dict(network.__salt__, {"cmd.run": mock_cmd}):
+                routes = network.routes(family="inet")
+                assert len(routes) == len(expected_routes)
+                for route in expected_routes:
+                    assert route in routes
+
+
+def test_netstat_route_linux_ipv4(expected_routes):
+    """
+    Tests that _netstat_route_linux can parse ipv4 routes appropriately
+    """
+    netstat_output = [
+        "0.0.0.0         100.64.24.145   0.0.0.0         UG        0 0          0 eth0",
+        "100.64.24.144   0.0.0.0         255.255.255.240 U         0 0          0 eth0",
+        "169.254.0.0     0.0.0.0         255.255.0.0     U         0 0          0 eth0",
+        "192.168.0.0     100.64.24.145   255.255.255.0   UG        0 0          0 eth0",
+        "192.168.1.0     100.64.24.145   255.255.255.0   UG        0 0          0 eth0",
```

Comment:
**question** Am I misunderstanding something? https://github.com/saltstack/salt/issues/58377 has the following:

```
default proto bird metric 32 
	nexthop via 1.2.3.4 dev eth1 weight 1 
	nexthop via 2.3.4.5 dev eth2 weight 1 
```

Which doesn't look like anything that's being tested for. Is that an incorrect understanding?

Reply:
It's been awhile since I looked at this, but I believe when I pasted in the output from that code into my issue, it parsed properly without any errors. 

I didn't add a test for that specific issue, but the core of the issue was poor parsing of ip route output. This code should be much more resilient and less prone to misalignment or keyword mismatches.

Reply:
Hrm. Well, I tried adding the output to both of these tests and they both failed. I even tried providing the output from [this comment](https://github.com/saltstack/salt/issues/58377#issuecomment-800466893) in the ip_route and I got the exact same output in the bug report.

```
    ip_route_output = [
        "::1 dev lo proto kernel metric 256 pref medium",
        "fe80::/64 dev xxx proto kernel metric 100 pref medium",
        "local ::1 dev lo table local proto kernel metric 0 pref medium",
        "local fe80::250:xxxx:xxxx:xxxx dev xxx table local proto kernel metric 0 pref medium",
        "multicast ff00::/8 dev xxx table local proto kernel metric 256 pref medium",
    ]
```
that gives me `ValueError: 'local' does not appear to be an IPv4 or IPv6 network`

and 

```
    ip_route_output = [
         "default proto bird metric 32",
         "   nexthop via 1.2.3.4 dev eth1 weight 1",
         "   nexthop via 2.3.4.5 dev eth2 weight 1",
    ]
```

gives me the same error but `nexthop`.

It's entirely possible that I'm doing it wrong, as I didn't spend an awful lot of time to really grok what the tests/changes are supposed to do, but I'd definitely like to see some tests showing that the mentioned issue is actually resolved.